sound/soc/sof/sof-acpi-dev.c

```diff
@@ -117,6 +118,39 @@ static const struct sof_dev_desc sof_acpi_cherrytrail_desc = {
 	.arch_ops = &sof_xtensa_arch_ops
 };
 
+static int minnowboard_quirk_cb(const struct dmi_system_id *id)
+{
+	struct device *dev = id->driver_data;
+
+	/*
+	 * Increment the usage count so as to prevent the device
+	 * from entering runtime suspend.
+	 */
+	pm_runtime_get_noresume(dev);
```

Comment:
btw what happens if instead of taking a fake reference you just don't enable pm_runtime?

Reply:
@plbossart this is what I want to do ideally. But the one test that fails with this change is the ipc_flood_test(). When this test runs, we call pm_runtime_get_sync() to resume the DSP if it is suspended and this throws an error is runtime_pm is not enabled.
So if I add a check to see if runtime PM is enabled before calling get_sync(), then it would solve the problem. And of course, every other place we add a pm_runtime_get_sync() would need to the same too. What do you think?